eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -93,38 +107,41 @@ filter Remove-PurgeableResources {
   $subscriptionId = (Get-AzContext).Subscription.Id
 
   foreach ($r in $Resource) {
+    Log "Attempting to purge $($r.AzsdkResourceType) '$($r.AzsdkName)'"
     switch ($r.AzsdkResourceType) {
       'Key Vault' {
-        Log "Attempting to purge $($r.AzsdkResourceType) '$($r.VaultName)'"
         if ($r.EnablePurgeProtection) {
           # We will try anyway but will ignore errors
           Write-Warning "Key Vault '$($r.VaultName)' has purge protection enabled and may not be purged for $($r.SoftDeleteRetentionInDays) days"
         }
 
-        Remove-AzKeyVault -VaultName $r.VaultName -Location $r.Location -InRemovedState -Force -ErrorAction Continue
+        # Using `$($using:r)` in the `-ScriptBlock` to make sure `$r` is captured for jobs.
+        Wait-PurgeableResource -Resource $r -Timeout $Timeout -PassThru:$PassThru -ScriptBlock { Remove-AzKeyVault -VaultName $($using:r).VaultName -Location $($using:r).Location -InRemovedState -Force -ErrorAction Continue }
```

Comment:
nit: just for readability it might be worth splitting the script block onto a newline like you did with the other case.

Reply:
I could. Originally it was because I was also logging (as warning) the failed script and this one-liner looked nicer, but I'm no longer doing that. That said, I may only do it if there's any further changes and...I really hope not. This has taken much more time than I anticipated, and not even just the execution of it. The closure issue was a real head-scratcher.

Reply:
Done